CMakeLists.txt

```diff
@@ -1355,23 +1363,6 @@ if(HEX_FILES_TO_MERGE)
   list(APPEND FLASH_DEPS mergehex)
 endif()
 
-if(CONFIG_OUTPUT_PRINT_MEMORY_USAGE)
-  # Use --print-memory-usage with the first link.
-  #
-  # Don't use this option with the second link because seeing it twice
-  # could confuse users and using it on the second link would suppress
-  # it when the first link has a ram/flash-usage issue.
-  set(option ${LINKERFLAGPREFIX},--print-memory-usage)
-  string(MAKE_C_IDENTIFIER check${option} check)
-
-  set(SAVED_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
-  set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} ${option}")
-  zephyr_check_compiler_flag(C "" ${check})
```

Comment:
I don't see the need for the abstraction.

If the compiler does not support --print-memory-usage, then this check will fail and it will not be added.

I would prefer to avoid the abstraction because it is very poor, there are a lot of details/assumptions leaking through it's API.

Reply:
The abstraction is needed to enable toolchains, where the linker does not support the --print-memory-usage flag, to implement a solution providing similar output based upon the CONFIG_OUTPUT_PRINT_MEMORY_USAGE configuration option.
Without this abstraction, other toolchains will not be able to provide memory usage information per above mentioned configuration option, hence the need for the abstraction of the print memory usage and the binary tools abstraction foundation.

Reply:
I see